### PR TITLE
SDIT-1734 create sentence initial commit

### PR DESCRIPTION
**Pull request by lbennett-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt [hunk 1]
```diff
@@ -940,3 +1016,35 @@ data class ExistingOffenderChargeRequest(
...
+  // either I or A
+  val status: String = "A",
+  // 1967, 1991, 2003, 2020
+  val sentenceCategory: String,
+  // eg ADIMP_ORA
+  val sentenceCalcType: String,
+  // 'IND' or 'AGG'
+  val sentenceLevel: String,
+  val fine: BigDecimal? = null,
+  // the prototype implies only 1 appearance can be associated with the case on creation
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt:1033]: will remove this comment in next pr

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt [hunk 2]
```diff
@@ -415,6 +424,39 @@ class SentencingService(
...
+          startDate = request.startDate,
+          endDate = request.endDate,
+          status = request.status,
+          fineAmount = request.fine,
+          sentenceLevel = request.sentenceLevel,
+        ),
+      )
+
+      CreateSentenceResponse(
+        sentenceSeq = sentence.id.sequence,
```
Comment (lbennett-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt:446]: will be returning more created nested ids in further PRs

**Review by steverendell — APPROVED** [review]
lgtm

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt [hunk 3]
```diff
@@ -246,6 +246,76 @@ class SentencingResource(private val sentencingService: SentencingService) {
     bookingId: Long,
   ): SentenceResponse = sentencingService.getOffenderSentence(sequence, bookingId)
 
+  @PreAuthorize("hasRole('ROLE_NOMIS_SENTENCING')")
+  @PostMapping("/prisoners/{offenderNo}/sentencing")
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResource.kt:250]: very minor but should it be "sentences" in the url

**Pull request merged**